modules/reactor/doc/content/source/meshgenerators/ControlDrumMeshGenerator.md

```diff
@@ -0,0 +1,73 @@
+
++ the assembly type ID and is named "outer_assembly_<assembly_type_id>" (for example a control drum with an assembly type ID of 1 will have a boundary ID of 2001 and boundary name of "outer_assembly_1").
+
+If the assembly is extruded to three dimensions the top-most boundary ID must be assigned using [!param](/Mesh/ReactorMeshParams/top_boundary_id) and will have the name "top", while the bottom-most boundary must be assigned using [!param](/Mesh/ReactorMeshParams/bottom_boundary_id) and will have the name "bottom".
+
+## Metadata Information
+
+Users may be interested in defining additional metadata to represent the reactor geometry and region IDs assigned to each geometry zone, which may be useful to users who want mesh geometry and composition information without having to inspect the generated mesh itself. The following metadata is defined on the control drum mesh:
+
+- `assembly_type`: Value of type_id associated with control drum, equivalent to the input parameter [!param](/Mesh/ControlDrumMeshGenerator/assembly_type)
+- `pitch`: Assembly pitch, equivalent to the input parameter [!param](/Mesh/ReactorMeshParams/assembly_pitch)
+- `is_control_drum`: Whether or not this mesh is a control drum, equal to true for all structures created by `ControlDrumMeshGenerator`.
+- `drum_radii`: Vector of length two corresponding to the inner and outer radii of the drum region, controlled by [!param](/Mesh/ControlDrumMeshGenerator/drum_inner_radius) and [!param](/Mesh/ControlDrumMeshGenerator/drum_outer_radius), respectively.
+- `drum_region_ids`: 2-D vector of region ids corresponding to radial and axial zones within control drum regions of assembly mesh, equivalent to the input parameter [!param](/Mesh/ControlDrumMeshGenerator/region_ids). Inner indexing is radial zones, while outer index is axial zones.
+
+In addition, the value of the `reactor_params_name` metadata can be used to retrieve global metadata defined by [ReactorMeshParams](ReactorMeshParams.md). Please refer to [ReactorMeshParams](ReactorMeshParams.md) to see a list of metadata defined by this mesh generator.
+
+For applications where an output mesh does not need to be created and meshing routines can consist entirely of defining reactor-based metadata, the parameter `[Mesh]`/[!param](/Mesh/MeshGeneratorMesh/data_driven_generator) can be set to the mesh generator that would generate an output mesh from RGMB metadata.
+
+## Example Syntax
+
+!listing modules/reactor/test/tests/meshgenerators/control_drum_mesh_generator/drum_pad.i block=Mesh
+
+If [!param](/Mesh/ControlDrumMeshGenerator/pad_start_angle) and [!param](/Mesh/ControlDrumMeshGenerator/pad_end_angle) are not provided, the drum region is discretized with the same region ID applied to all azimuthal drum elements. In this case, only 3 values per axial level should be provided in [!param](/Mesh/ControlDrumMeshGenerator/region_ids) (drum inner, drum, background).
+
```

Comment:
could point to the function that provides the actual pad fraction here?

Reply:
What function is this? I'm not familiar with it

Reply:
https://mooseframework.inl.gov/source/functions/MultiControlDrumFunction.html

Reply:
Added reference to MultiControlDrumFunction